libraries/standard/mqtt/src/iot_mqtt_api.c

```diff
@@ -1600,12 +1547,10 @@ IotMqttError_t IotMqtt_PublishAsync( IotMqttConnection_t mqttConnection,
     _mqttOperation_t * pOperation = NULL;
     uint8_t ** pPacketIdentifierHigh = NULL;
 
-    /* Default PUBLISH serializer function. */
-    IotMqttError_t ( * serializePublish )( const IotMqttPublishInfo_t *,
-                                           uint8_t **,
-                                           size_t *,
-                                           uint16_t *,
-                                           uint8_t ** ) = _IotMqtt_SerializePublish;
+    /* Choose the appropriate PUBLISH serializer function. */
+    _publishSerializer_t serializePublish = _getMqttPublishSerializer(
+        mqttConnection->pSerializer
```

Comment:
mqttConnection might not have been initialized here either.

Reply:
This has been refactored away